### DISCUSSION TITLE
[A-L] (20 Settembre 2024) Esercizio 5

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#421] [Esami A-L]
- Tizio dice: Caio e Sempronio non sono entrambi ladri.

- Caio dice: Sempronio è un ladro se Tizio non è un ladro.

- Sempronio dice: Non sono un ladro e almeno uno degli altri due è un ladro.

---

1. Formulare le tre testimonianze in un adeguato linguaggio proposizionale (dichiarando il significato delle variabili scelte).
    [1 punto]

2. È possibile che tutti e tre dicano il vero? (Suggerimento: scrivere la tavola di verità) [2 punti]

3. Assumendo che chi è un ladro ha mentito e chi non è un ladro ha detto il vero, si può stabilire chi è un ladro e chi no? (Giustificare brevemente la risposta). [2 punti]

**Comment by DavideGalilei** [comment]
### 1. Linguaggio proposizionale
È possibile formulare le testimonianze con tre variabili $$\large{T, C, S}$$, dove la variabile ha valore $\large{0}$ se la persona è innocente, e valore $\large{1}$ se la persona è un ladro.
- Tizio dice: $\large{\neg{(C \land S)}} = \neg{C} \lor \neg{S}$ (secondo De Morgan)
- Caio dice: $\large{\neg{T} \to S} = T \lor S$ (secondo la tabella di verità dell'operatore implicazione $\large{\to}$)
- Sempronio dice: $\large{\neg{S} \land (C \lor T)}$

<img src="https://github.com/user-attachments/assets/18001e59-f1c3-49ae-9bdf-b5d7311c5b95" width="350">

### 2. È possibile che tutti e tre dicano il vero?
Per determinare se tutti e tre dicano il vero, è necessario stendere una tabella di verità "forzando" i ruoli di innocente/ladro a Tizio, Caio e Sempronio, e controllando se le loro affermazioni siano coerenti tra di loro secondo i valori precedentemente imposti dei loro ruoli.


| $\large{T}$ | $\large{C}$ | $\large{S}$ | $\large{\neg(C \wedge S)}$ | $\large{T \lor S}$ | $\large{\neg S \wedge (T \lor C)}$ | Tutte vere? |
|:----:|:----:|:----:|:-----------------------:|:--------------:|:---------------------------:|:-----------:|
| 0 | 0 | 0 | 1 | 0 | 0 | No  |
| 0 | 0 | 1 | 1 | 1 | 0 | No  |
| 0 | 1 | 0 | 1 | 0 | 1 | No  |
| 0 | 1 | 1 | 0 | 1 | 0 | No  |
| **1** | **0** | **0** | **1** | **1** | **1** | **Sì** |
| 1 | 0 | 1 | 1 | 1 | 0 | No  |
| **1** | **1** | **0** | **1** | **1** | **1** | **Sì** |
| 1 | 1 | 1 | 0 | 1 | 0 | No  |

<img src="https://github.com/user-attachments/assets/725b9c19-eca6-4cf9-9cc7-2c89381d9431" width="200">

Dalla tabella emergono due casi dove le affermazioni risultano tutte coerenti tra di loro, ovvero $\large{\text{AffermazioneTizio} \land \text{AffermazioneCaio} \land \text{AffermazioneSempronio} = \neg{(C \land S)} \land (\neg{T} \to S) \land \neg{S} \land (C \lor T) = \text{SAT}}$

Dunque la risposta è: ✅ **VERO**.

### 3. Regola dei ladri
La regola dei ladri ci dice:
> chi è un ladro ha mentito e chi non è un ladro ha detto il vero

ciò significa che nella tabella di verità dovremmo andare ad invertire le affermazioni dei ladri.

Ad esempio nel caso imposto $(T, C, S) = (0, 0, 1)$ il precedente verdetto della validità delle affermazioni era $(1, 1, 0)$, ma nella nuova tabella della verità seguendo la regola dei ladri ciò diventerebbe $(1, 1, 1)$, in quanto Sempronio (S, nella terza posizione) è stato imposto come ladro e quindi la sua opinione va invertita.

| $\large{T}$ | $\large{C}$ | $\large{S}$ | $\large{\neg(C \wedge S)}$ originale | $\large{T \lor S}$ originale | $\large{\neg S \wedge (T \lor C)}$ originale | $\large{\neg(C \wedge S)}$ reale | $\large{T \lor S}$ reale | $\large{\neg S \wedge (T \lor C)}$ reale |
|:----:|:----:|:----:|:-----------------------:|:--------------:|:---------------------------:|:-----------------------:|:--------------:|:---------------------------:|
| 0 | 0 | 0 | 1 | 0 | 0 | 1 | 0 | 0 |
| 0 | 0 | 1 | 1 | 1 | 0 | 1 | 1 | $\underline{1}$ |
| 0 | 1 | 0 | 1 | 0 | 1 | 1 | $\underline{1}$ | 1 |
| 0 | 1 | 1 | 0 | 1 | 0 | 0 | $\underline{0}$ | $\underline{1}$ |
| 1 | 0 | 0 | 1 | 1 | 1 | $\underline{0}$ | 1 | 1 |
| 1 | 0 | 1 | 1 | 1 | 0 | $\underline{0}$ | 1 | $\underline{1}$ |
| 1 | 1 | 0 | 1 | 1 | 1 | $\underline{0}$ | $\underline{0}$ | 1 |
| 1 | 1 | 1 | 0 | 1 | 0 | $\underline{1}$ | $\underline{0}$ | $\underline{1}$ |

<img src="https://github.com/user-attachments/assets/398f6d60-2d72-4466-8084-c2c1cb636c47" width="600">

---

Prendiamo i casi in cui non si verificano incongruenze tra le affermazioni di Tizio, Caio e Sempronio applicando la regola dei ladri (cioè, se uno è ladro, invertiamo la sua opinione per valutarne la coerenza).

Nei casi $3.1$ e $3.2$, applicando la regola dei ladri, otteniamo due diverse configurazioni che risultano entrambe coerenti. Ma osservando i valori per $\large{T,C,S}$ possiamo vedere che ci sono due casi $\large{(0, 0, 1)}$ e $\large{(0, 1, 0)}$ dove $\large{C}$ e $\large{S}$ hanno valori rispettivamente $\large{(0, 1)}$ e $\large{(1, 0)}$, ciò significa che in nonostante le due configurazioni siano coerenti secondo le affermazioni, non è possibile determinare chi tra Caio e Sempronio sia il ladro, perché sono valide sia la configurazione dove Caio è ladro e Sempronio innocente, e quella dove Caio è innocente e Sempronio ladro.

Di conseguenza, la risposta è ❌ **FALSO**:
Non possiamo determinare con certezza chi è ladro tra Caio e Sempronio.

Il fatto che esistano due configurazioni diverse ma entrambe valide (applicando la regola dei ladri) ci impedisce di identificare univocamente chi è il colpevole, anche se **sappiamo con certezza che Tizio è innocente**, questo perché in entrambe le configurazioni coerenti secondo la regola dei ladri la variabile $\large{T}$ (Tizio) ha sempre valore $\large{0}$!
